### PR TITLE
Adapt class Person to Student to fit with TAchy's model design

### PR DESCRIPTION
**Pull request by jian7490**

### CODE REVIEW

File: src/test/java/seedu/address/testutil/PersonUtil.java [hunk 1]
```diff
@@ -9,46 +8,44 @@
 import java.util.Set;
 
 import seedu.address.logic.commands.AddCommand;
-import seedu.address.logic.commands.EditCommand.EditPersonDescriptor;
-import seedu.address.model.person.Person;
+import seedu.address.logic.commands.EditCommand;
```
Comment (lm-44) [inline src/test/java/seedu/address/testutil/PersonUtil.java:11]: Maybe import EditCommand.EditStudentDescriptor?

File: src/test/java/seedu/address/testutil/PersonUtil.java [hunk 2]
```diff
@@ -9,46 +8,44 @@
...
         );
         return sb.toString();
     }
 
     /**
-     * Returns the part of command string for the given {@code EditPersonDescriptor}'s details.
+     * Returns the part of command string for the given {@code EditStudentDescriptor}'s details.
      */
-    public static String getEditPersonDescriptorDetails(EditPersonDescriptor descriptor) {
+    public static String getEditPersonDescriptorDetails(EditCommand.EditStudentDescriptor descriptor) {
```
Comment (lm-44) [inline src/test/java/seedu/address/testutil/PersonUtil.java:44]: With the above import changed, you can remove "EditCommand."

**Review by lm-44 — CHANGES_REQUESTED** [review]
I've tried my very best to go through the code HAHAHA. My eyes hurt. I think majority of it is fine, just some nits. Other than that LGTM!

File: src/main/java/seedu/address/logic/commands/exceptions/CommandException.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package seedu.address.logic.commands.exceptions;
 
+import seedu.address.logic.commands.Command;
+
 /**
  * Represents an error which occurs during execution of a {@link Command}.
```
[the remark this answers is not in the document]
Comment (jian7490) [reply src/main/java/seedu/address/logic/commands/exceptions/CommandException.java:6]: Sure. Good advice to me on improving  the code quality!

File: src/main/java/seedu/address/model/student/Student.java [hunk 4]
```diff
@@ -11,29 +11,26 @@
...
  */
-public class Person {
+public class Student {
 
     // Identity fields
     private final Name name;
     private final Phone phone;
     private final Email email;
 
-    // Data fields
```
[the remark this answers is not in the document]
Comment (jian7490) [reply src/main/java/seedu/address/model/student/Student.java:24]: Solid point. I believe it was removed by accident.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 5]
```diff
@@ -102,29 +99,6 @@ public void parsePhone_validValueWithWhitespace_returnsTrimmedPhone() throws Exc
...
-    @Test
-    public void parseAddress_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseAddress((String) null));
-    }
-
-    @Test
-    public void parseAddress_invalidValue_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parseAddress(INVALID_ADDRESS));
-    }
-
```
Comment (emmannyyy) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:114]: If we do intend on keeping address, then we should recover these tests. Let's discuss with the team if keeping address is the right way to go

File: src/test/java/seedu/address/model/person/AddressTest.java
Comment (emmannyyy) [inline src/test/java/seedu/address/model/person/AddressTest.java:1]: Same, may want to keep this test

File: src/test/java/seedu/address/storage/JsonAdaptedStudentTest.java [hunk 6]
```diff
@@ -12,98 +12,80 @@
...
-        JsonAdaptedPerson person =
-                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, INVALID_ADDRESS, VALID_TAGS);
-        String expectedMessage = Address.MESSAGE_CONSTRAINTS;
-        assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
-    }
-
-    @Test
-    public void toModelType_nullAddress_throwsIllegalValueException() {
-        JsonAdaptedPerson person = new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, null, VALID_TAGS);
-        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Address.class.getSimpleName());
```
Comment (emmannyyy) [inline src/test/java/seedu/address/storage/JsonAdaptedStudentTest.java:97]: same here, may have to recover this

**Pull request merged**